bazaar/src/tests.rs

```diff
@@ -53,261 +51,213 @@ impl ExtBuilder {
+
+        assert_eq!(EncointerBazaar::offering_registry(BusinessIdentifier::new(cid, alice()), 1),
+                   OfferingData::new(url1()));
+    });
+}
+
+#[test]
+fn update_inexistent_offering_is_err() {
+    ExtBuilder::build().execute_with(|| {
+        let cid = create_cid();
+        BusinessRegistry::<TestRuntime>::insert(cid, alice(), BusinessData::new(url(), 2));
+        OfferingRegistry::<TestRuntime>::insert(BusinessIdentifier::new(cid, alice()), 1,
+                                                OfferingData::new(url()));
+
+        assert!(EncointerBazaar::update_offering(Origin::signed(bob()), cid, 1, url1()).is_err());
+        assert!(EncointerBazaar::update_offering(Origin::signed(alice()), cid, 0, url1()).is_err());
+        assert!(EncointerBazaar::update_offering(Origin::signed(alice()), CommunityIdentifier::zero(), 1, url1()).is_err());
+    });
+}
+
+#[test]
+fn delete_existing_offering_is_ok() {
+    ExtBuilder::build().execute_with(|| {
+        let cid = create_cid();
+        BusinessRegistry::<TestRuntime>::insert(cid, alice(), BusinessData::new(url(), 2));
+        OfferingRegistry::<TestRuntime>::insert(BusinessIdentifier::new(cid, alice()), 1,
+                                                OfferingData::new(url()));
+
+        assert!(EncointerBazaar::delete_offering(Origin::signed(alice()), cid, 1).is_ok());
+
+        assert_eq!(EncointerBazaar::offering_registry(BusinessIdentifier::new(cid, alice()), 1),
+                   OfferingData::default());
+    });
+}
+
+#[test]
+fn delete_inexistent_offering_is_err() {
+    ExtBuilder::build().execute_with(|| {
+        let cid = create_cid();
+        BusinessRegistry::<TestRuntime>::insert(cid, alice(), BusinessData::new(url(), 2));
+        OfferingRegistry::<TestRuntime>::insert(BusinessIdentifier::new(cid, alice()), 1,
+                                                OfferingData::new(url()));
+
+        assert!(EncointerBazaar::delete_offering(Origin::signed(bob()), cid, 1).is_err());
+        assert!(EncointerBazaar::delete_offering(Origin::signed(alice()), cid, 0).is_err());
+        assert!(EncointerBazaar::delete_offering(Origin::signed(alice()), CommunityIdentifier::zero(), 1).is_err());
+    });
+}
+
+#[test]
+fn when_deleting_business_delete_all_its_offerings() {
+    ExtBuilder::build().execute_with(|| {
+        let cid = create_cid();
+        BusinessRegistry::<TestRuntime>::insert(cid, alice(), BusinessData::new(url(), 2));
+        BusinessRegistry::<TestRuntime>::insert(cid, bob(), BusinessData::new(url1(), 2));
+        OfferingRegistry::<TestRuntime>::insert(BusinessIdentifier::new(cid, alice()), 1, OfferingData::new(url()));
+        OfferingRegistry::<TestRuntime>::insert(BusinessIdentifier::new(cid, bob()), 1, OfferingData::new(url1()));
+
+        assert!(EncointerBazaar::delete_business(Origin::signed(alice()), cid).is_ok());
+
+        assert_eq!(EncointerBazaar::business_registry(cid, alice()), BusinessData::default());
```

Comment:
hmm, isn't there a way to test for null specifically?

Reply:
we coud check for it by directly acessing the double_storage_map (i.e. accessing internal state) and checking for "contains"